### PR TITLE
Fix crash when drawing outside top canvas border

### PR DESCRIPTION
**Pull request by humdingerb**
Only 64bit build was suffering that crash apparently.
Patch provided by nephele.

### CODE REVIEW

File: WonderBrush/src/tools/PenStroke.cpp [hunk 1]
```diff
@@ -383,6 +383,8 @@ PenStroke::_StrokePoint(float x, float y, float pressure, uint8* bits, uint32 bp
 	else
 		pressure = fAlpha.max;
 	uint8 maxAlpha = (uint8)(255.0 * pressure);
+	if (y < 0)
+		return;
```
Comment (stippi) [inline WonderBrush/src/tools/PenStroke.cpp:387]: This doesn't seem right. From what I remember, `x` and `y` specify the center of the shape that is to be rendered by the `PenStroke`. The `constrainRect` should contain the necessary clipping. If somehow it is possible to render outside the bitmap buffer, then the bug is certainly not at this place. And if it were, then we should just as well check for `x < 0` and also right and bottom buffer bounds.

**Review by stippi — CHANGES_REQUESTED** [review]

**Comment by stippi** [comment]
I can reproduce the bug. Looking into it.

**Comment by stippi** [comment]
Fixed by https://github.com/stippi/WonderBrush-v2/commit/b8c8f0d9cba3d194cb32dcbac41960a4b113c83b

**Comment by humdingerb** [comment]
Thanks for the quick investigation and fix! Should I update the haikuports recipe to publish the fixed version, or do you have other commits lined up that should be merged first?

**Comment by stippi** [comment]
I have nothing else lined up for WB 2, sorry. Publishing this fixed version is much appreciated!

**Comment by humdingerb** [comment]
I'll wait on your thoughts on #27. :)

**Pull request closed without merging**